internal/pipeline/pipeline.go

```diff
@@ -0,0 +1,55 @@
+package pipeline
+
+import (
+	"fmt"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/domain/status"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/usecases/link"
+	"net/http"
+	"sync"
+	"time"
+)
+
+func LinkStatusUpdater(luc *link.LinkUseCases) {
+	go func() {
+		links := make(chan link.Link)
```

Comment:
The idea behind the pipeline was to have ready jobs for all the workers at the next stage. 
So, you have four workers at the next stage. The first one gets its job but the others are waiting for the producer to generate more new jobs. 

And the next point is about overall efficiency. Why do you have 4 (hardcoded) workers if the first one gets almost all jobs while the others will have almost none?

Reply:
We created a buffered channel, it should help at this point. However, our previous solution seems acceptable too, since each worker processes quite complicated requests: an http request and a request to the database.